core/node/state_keeper/src/io/seal_logic/l2_block_seal_subtasks.rs

```diff
@@ -333,10 +333,9 @@ impl L2BlockSealSubtask for InsertTokensSubtask {
         connection: &mut Connection<'_, Core>,
     ) -> anyhow::Result<()> {
         let is_fictive = command.is_l2_block_fictive();
-        let mut deployer_address = command.l2_native_token_vault_proxy_addr;
-        if command.l2_legacy_shared_bridge_addr != H160::zero() {
-            deployer_address = command.l2_legacy_shared_bridge_addr;
-        }
+        let deployer_address = command
+            .l2_shared_bridge_addr
```

Comment:
This is not exactly correct... Since for chains without legacy bridge there is a shared bridge (the l2 asset router), but no NTV

Reply:
I see where the confusion comes from. In my understanding L2 asset router is not a shared bridge, it's L2 asset router. Shared bridge is only `L2SharedBridge`. IMO current state is good enough (operator doesn't need to do any config changes, `deployer_address` and `l2_shared_bridge` in API will return correct values automatically. `deployer_address` for EN is a bit ugly as I had to add another field in API response)